### ISSUE TITLE
hokusai version command shows dev version, when installed via Pip

### ISSUE THREAD

**Issue by artsyjian** [artsy/hokusai#357]
Install Hokusai via Pip:

```
pip install hokusai
```

Hokusai version command shows a dev version:
```
$ hokusai version
1.1.1.dev0+gb6a5062.d20230419
```

`_version.py` (generated by setuptools_scm) indeed has dev version:
```
jian@artsy:~/.pyenv/versions/3.9.10/lib/python3.9/site-packages/hokusai$ cat _version.py
# file generated by setuptools_scm
# don't change, don't track in version control
__version__ = version = '1.1.1.dev0+gb6a5062.d20230419'
__version_tuple__ = version_tuple = (1, 1, 1, 'dev0', 'gb6a5062.d20230419')
```

The Makefile's [publish-to-pip step](https://github.com/artsy/hokusai/blob/main/Makefile#L96) runs `setuptools_scm` which generates the canonical version (1.1.0). The problem is with the subsequent commands:
- `poetry version`, which bumps the [version stored in pyproject.toml](https://github.com/artsy/hokusai/blob/d4a9714779298d6348ba25a188eb3f9ab6bd65c1/pyproject.toml#L6), to `1.1.0`.
- `poetry build`, which actually runs `setuptools_scm` too, and this time, because of the version bump in pyproject.toml, generates a dev version.

For example, if we were to release v1.2.0:
```
(hokusai) jian@artsy:~/code/hokusai$ python -m setuptools_scm
1.2.0

(hokusai) jian@artsy:~/code/hokusai$ cat hokusai/_version.py
# file generated by setuptools_scm
# don't change, don't track in version control
__version__ = version = '1.2.0'
__version_tuple__ = version_tuple = (1, 2, 0)

(hokusai) jian@artsy:~/code/hokusai$ poetry version 1.2.0
Bumping version from 0.0.0 to 1.2.0

(hokusai) jian@artsy:~/code/hokusai$ poetry build
Building hokusai (1.2.0)
  - Building sdist
  - Built hokusai-1.2.0.tar.gz
  - Building wheel
  - Built hokusai-1.2.0-py3-none-any.whl

(hokusai) jian@artsy:~/code/hokusai$ cat hokusai/_version.py
# file generated by setuptools_scm
# don't change, don't track in version control
__version__ = version = '1.2.1.dev0+gb6a5062.d20230703'
__version_tuple__ = version_tuple = (1, 2, 1, 'dev0', 'gb6a5062.d20230703')
```

**Comment by artsyjian** [comment]
If the first `setuptools_scm` run is omitted, and `poetry version` is omitted, `poetry build` does result in correct `hokusai/_version.py`:
```
(hokusai) jian@artsy:~/code/hokusai$ git tag v1.2.0

(hokusai) jian@artsy:~/code/hokusai$ poetry build
Building hokusai (0.0.0)
  - Building sdist
  - Built hokusai-0.0.0.tar.gz
  - Building wheel
  - Built hokusai-0.0.0-py3-none-any.whl

(hokusai) jian@artsy:~/code/hokusai$ cat hokusai/_version.py
# file generated by setuptools_scm
# don't change, don't track in version control
__version__ = version = '1.2.0'
__version_tuple__ = version_tuple = (1, 2, 0)
```

However, the version in archive's filename is wrong.

If `poetry version` is run (so that the archive's name is correct), `hokusai/_version.py` is again incorrect:
```
(hokusai) jian@artsy:~/code/hokusai$ git tag v1.2.0

(hokusai) jian@artsy:~/code/hokusai$ poetry version 1.2.0
Bumping version from 0.0.0 to 1.2.0

(hokusai) jian@artsy:~/code/hokusai$ poetry build
Building hokusai (1.2.0)
  - Building sdist
  - Built hokusai-1.2.0.tar.gz
  - Building wheel
  - Built hokusai-1.2.0-py3-none-any.whl

(hokusai) jian@artsy:~/code/hokusai$ cat hokusai/_version.py
# file generated by setuptools_scm
# don't change, don't track in version control
__version__ = version = '1.2.1.dev0+gb6a5062.d20230703'
__version_tuple__ = version_tuple = (1, 2, 1, 'dev0', 'gb6a5062.d20230703')
```

**Comment by artsyjian** [comment]
resolved by #389
